### DISCUSSION TITLE
Unpublished all objects of a classe

### DISCUSSION THREAD

**Discussion by ghost** [pimcore/pimcore#10071] [Q&A / Support] [answered]
How can I unpublish all objects of a classe at once?

Edit : Sorry I forgot to specify: in PHP with the API

**Comment by NiklasBr** [comment, marked as answer]
1. Click on *Home*:
![image](https://user-images.githubusercontent.com/279826/129913420-654ab022-fc1a-44cb-989e-86c85cf6eb69.png)

2. Choose the object type:
![image](https://user-images.githubusercontent.com/279826/129913502-fc9171f8-afe7-47b5-9419-e9a40fd41e1f.png)

3. In the *Published* column, click the small down arrow and choose *Batch edit all*:
![image](https://user-images.githubusercontent.com/279826/129913693-97e8f535-96a1-4329-bdbe-d2982abdcf0e.png)

**Reply by ghost** [reply]
Sorry I forgot to specify: in PHP with the API

**Reply by NiklasBr** [reply]
```php
foreach (new YourObjectClass\Listing() as $object) {
    $object->setPublished(false);
    $object->save();
}
```

Maybe you also want to turn off versions for this operation: https://pimcore.com/docs/pimcore/current/Development_Documentation/Tools_and_Features/Versioning.html#page_Turn-off-Versioning-for-the-Current-Process
